docs/how-tos/build-a-plugin.md

```diff
@@ -0,0 +1,385 @@
+# Build your own pynxtools plugin
+
+Your current data is not supported yet by the [built-in pynxtools readers](https://github.com/FAIRmat-NFDI/pynxtools/tree/master/src/pynxtools/dataconverter/readers) or any of the officially supported [pynxtools plugins](../reference/plugins.md)? 
+
+Don't worry, the following how-to will guide you how to write a reader your own data.
+
+
+## Getting started
+You should start by creating a clean repository that implements the following structure (for a plugin called ```pynxtools-plugin```):
+```
+pynxtools-plugin
+├── .github/workflows
+├── docs
+│   ├── explanation
+│   ├── how-tos
+│   ├── reference
+│   ├── tutorial
+├── src
+│   ├── pynxtools_plugin
+│       ├── reader.py
+├── tests
+│   └── data
+├── LICENSE
+├── mkdocs.yaml
+├── dev-requirements.txt
+└── pyproject.toml
+```
+
+To identify `pynxtools-plugin` as a plugin for pynxtools, an entry point must be established:
+```
+[project.entry-points."pynxtools.reader"]
+mydatareader = "pynxtools_plugin.reader:MyDataReader"
+```
+
+Here, we will focus mostly on the `reader.py` file and how to build a reader. For guidelines on how to build the other parts of your plugin, you can have a look here:
+- [Documentation writing guide](https://nomad-lab.eu/prod/v1/staging/docs/writing_guide.html)
+- [Plugin testing framework](using-pynxtools-test-framework.md)
+
+Note: There is also a [cookiecutter template](https://github.com/FAIRmat-NFDI/pynxtools-plugin-template) available for creating your own pynxtools plugin, but this is currently not well-maintained.
+
+
+## Writing a Reader
+
+The pynxtools converter allows extending support to other data formats by allowing extensions called readers.
+The converter provides a dev platform to build a NeXus compatible reader by providing checking against a chosen NeXus Application Definition.
+
+<img src="media/convert_routine.svg"/>
```

Comment:
I find the picture a bit overcomplicated and you can essentially tell it's content by writing. 

```
dataconverter --reader <reader> --nxdl <nxdl> files ...
```

Also it still uses deprecated syntax such as `--input-file`.

Reply:
I recommend to only state the command here and explain it.

Reply:
You can also connect it to the reader name `mydatareader` from above. That way it's clear how to call a reader from the entrypoint.

Reply:
I removed the image and refactored this part a bit.